# Amaç 

Bu çalışmadaki amacım wide formatta birleştirdiğim finansal verilerin başarılı bir şekilde veri tabanına aktarılması 
Veri tabanında az yer kaplayacak şekilde küçültülmesi 
Kopyalanmış, eksik vb bozuk verilerin tespiti ve düzenlenmesi yapılacaktır

In [144]:
import pandas as pd 
import os 

In [237]:
wide = pd.read_excel("./combined_wide_format.xlsx")
pd.set_option('display.max_columns', 350)
pd.set_option("display.max_rows",400)

In [146]:
#columns have space and other char lets delete them 
wide.columns = [col.strip().replace('\u200b', '') for col in wide.columns]
# lets see results 
wide.columns

Index(['Company', 'index', 'Dönen Varlıklar', 'Nakit ve Nakit Benzerleri',
       'Gayrimenkul Projeleri Kapsamında Açılan Nakit Hesapları',
       'Finansal Yatırımlar', 'Teminata Verilen Finansal Varlıklar',
       'Ticari Alacaklar', 'Finans Sektörü Faaliyetlerinden Alacaklar',
       'Türkiye Cumhuriyet Merkez Bankası Hesabı',
       ...
       'Diğer Kapsamlı Gelir', 'Azınlık Payı', 'Özkaynaklar',
       'Diğer Uzun Vadeli Finansal Borçlar', 'Uzun Vadeli Karşılıklar',
       'Diğer Uzun Vadeli Ticari Borçlar',
       'Cari Dönem Vergisiyle İlgili Yükümlülükler', 'Diğer Finansal Borçlar',
       'Cari Dönem Vergisiyle İlgili Yükümlülükler.1',
       'Satış Amaçlı Elde Tutulan Duran Varlıklara İlişkin Yükümlülükler'],
      dtype='object', length=309)

In [147]:
wide.head(10)

Company    index  Dönen Varlıklar  Nakit ve Nakit Benzerleri  \
0   ENERY   2024/9              NaN               1.053510e+09   
1   ENERY   2024/6              NaN               6.314942e+08   
2   ENERY   2024/3              NaN               5.336559e+09   
3   ENERY  2023/12              NaN               5.935501e+09   
4   ENERY   2023/9              NaN               3.798841e+09   
5   ENERY   2023/6              NaN               1.283178e+09   
6   ENERY   2023/3              NaN                        NaN   
7   ENERY  2022/12              NaN               2.479796e+09   
8   ENERY  2021/12              NaN               2.689702e+08   
9   ENERY  2020/12              NaN               2.433848e+08   

   Gayrimenkul Projeleri Kapsamında Açılan Nakit Hesapları  \
0                                                NaN         
1                                                NaN         
2                                                NaN         
3                                                NaN         
4                                                NaN         
5                                                NaN         
6                                                NaN         
7                                                NaN         
8                                                NaN         
9                                                NaN         

   Finansal Yatırımlar  Teminata Verilen Finansal Varlıklar  Ticari Alacaklar  \
0                  NaN                                  NaN      1.067704e+09   
1                  NaN                                  NaN      9.063402e+08   
2                  NaN                                  NaN      2.382966e+09   
3                  NaN                                  NaN      3.202226e+09   
4                  NaN                                  NaN      7.723428e+08   
5                  NaN                                  NaN      1.200150e+09   
6                  NaN                                  NaN               NaN   
7                  NaN                                  NaN      5.646616e+09   
8                  NaN                                  NaN      8.359033e+08   
9                  NaN                                  NaN      4.417910e+08   

   Finans Sektörü Faaliyetlerinden Alacaklar  \
0                                        NaN   
1                                        NaN   
2                                        NaN   
3                                        NaN   
4                                        NaN   
5                                        NaN   
6                                        NaN   
7                                        NaN   
8                                        NaN   
9                                        NaN   

   Türkiye Cumhuriyet Merkez Bankası Hesabı  Diğer Alacaklar  \
0                                       NaN     4.542064e+06   
1                                       NaN     4.656865e+06   
2                                       NaN     1.915186e+07   
3                                       NaN     2.074794e+07   
4                                       NaN     6.369779e+07   
5                                       NaN     2.756520e+07   
6                                       NaN              NaN   
7                                       NaN     5.593593e+07   
8                                       NaN     4.446199e+08   
9                                       NaN     5.759283e+06   

   Müşteri Sözleşmelerinden Doğan Varlıklar  \
0                                       NaN   
1                                       NaN   
2                                       NaN   
3                                       NaN   
4                                       NaN   
5                                       NaN   
6                                       NaN   
7                                       NaN   
8                                       NaN

In [148]:
wide.tail(10)

Company    index  Dönen Varlıklar  Nakit ve Nakit Benzerleri  \
2837   AKBNK   2018/9              NaN               7.853838e+10   
2838   AKBNK   2018/6              NaN               5.823308e+10   
2839   AKBNK   2018/3              NaN               4.783258e+10   
2840   AKBNK  2017/12              NaN               3.536318e+10   
2841   AKBNK   2017/9              NaN               3.490064e+10   
2842   AKBNK   2017/6              NaN               3.982363e+10   
2843   AKBNK   2017/3              NaN               3.326419e+10   
2844   AKBNK  2016/12              NaN               3.501228e+10   
2845   AKBNK   2016/9              NaN               3.466362e+10   
2846   AKBNK   2016/6              NaN               3.508406e+10   

      Gayrimenkul Projeleri Kapsamında Açılan Nakit Hesapları  \
2837                                                NaN         
2838                                                NaN         
2839                                                NaN         
2840                                                NaN         
2841                                                NaN         
2842                                                NaN         
2843                                                NaN         
2844                                                NaN         
2845                                                NaN         
2846                                                NaN         

      Finansal Yatırımlar  Teminata Verilen Finansal Varlıklar  \
2837                  NaN                                  NaN   
2838                  NaN                                  NaN   
2839                  NaN                                  NaN   
2840                  NaN                                  NaN   
2841                  NaN                                  NaN   
2842                  NaN                                  NaN   
2843                  NaN                                  NaN   
2844                  NaN                                  NaN   
2845                  NaN                                  NaN   
2846                  NaN                                  NaN   

      Ticari Alacaklar  Finans Sektörü Faaliyetlerinden Alacaklar  \
2837               NaN                                        NaN   
2838               NaN                                        NaN   
2839               NaN                                        NaN   
2840               NaN                                        NaN   
2841               NaN                                        NaN   
2842               NaN                                        NaN   
2843               NaN                                        NaN   
2844               NaN                                        NaN   
2845               NaN                                        NaN   
2846               NaN                                        NaN   

      Türkiye Cumhuriyet Merkez Bankası Hesabı  Diğer Alacaklar  \
2837                                       NaN              NaN   
2838                                       NaN              NaN   
2839                                       NaN              NaN   
2840                                       NaN              NaN   
2841                                       NaN              NaN   
2842                                       NaN              NaN   
2843                                       NaN              NaN   
2844                                       NaN              NaN   
2845                                       NaN              NaN   
2846                                       NaN              NaN   

      Müşteri Sözleşmelerinden Doğan Varlıklar  \
2837                                       NaN   
2838                                       NaN   
2839                                       NaN   
2840                                       NaN   
2841                                       NaN   
2842

In [149]:
wide.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2847 entries, 0 to 2846
Columns: 309 entries, Company to Satış Amaçlı Elde Tutulan Duran Varlıklara İlişkin Yükümlülükler
dtypes: float64(307), object(2)
memory usage: 6.7+ MB


# Bu proje veri tabanına eklenecek buradaki sorun 309 tane sütun olması ve bunların çoğunun boş olması. Bu veri tabanındaki işlemlerin yapılma süresini uzattığı gibi fazla yer kaplıyor
Bu verileri azaltmak veya yeni tablolar oluşturmak gerekiyor mu bunu incelemek lazım 

In [151]:
wide.count(wide.empty)

Company                                                             2847
index                                                               2847
Dönen Varlıklar                                                        0
Nakit ve Nakit Benzerleri                                           2640
Gayrimenkul Projeleri Kapsamında Açılan Nakit Hesapları               48
                                                                    ... 
Diğer Uzun Vadeli Ticari Borçlar                                      16
Cari Dönem Vergisiyle İlgili Yükümlülükler                            11
Diğer Finansal Borçlar                                                16
Cari Dönem Vergisiyle İlgili Yükümlülükler.1                          16
Satış Amaçlı Elde Tutulan Duran Varlıklara İlişkin Yükümlülükler      12
Length: 309, dtype: int64

In [152]:
print(wide.isnull().sum())

Company                                                                0
index                                                                  0
Dönen Varlıklar                                                     2847
Nakit ve Nakit Benzerleri                                            207
Gayrimenkul Projeleri Kapsamında Açılan Nakit Hesapları             2799
                                                                    ... 
Diğer Uzun Vadeli Ticari Borçlar                                    2831
Cari Dönem Vergisiyle İlgili Yükümlülükler                          2836
Diğer Finansal Borçlar                                              2831
Cari Dönem Vergisiyle İlgili Yükümlülükler.1                        2831
Satış Amaçlı Elde Tutulan Duran Varlıklara İlişkin Yükümlülükler    2835
Length: 309, dtype: int64


In [210]:
#tüm satırları boş olan sütünların isimlerini yazdır 
empty_columns = wide.columns[wide.isnull().sum() >=  wide[wide.columns[0]].count() -1 ]

empty_columns


Index(['Dönen Varlıklar', 'Duran Varlıklar', 'Kısa Vadeli Yükümlülükler',
       'Uzun Vadeli Yükümlülükler', 'Özkaynaklar',
       'Ortak Kontrole Tabi Teşebbüs veya İşletmeleri İçeren Birleşmelerin Etkisi',
       'Finansal Varlıklar (Net)', 'Unnamed: 1'],
      dtype='object')

In [212]:
for column_name in empty_columns :
    display(wide[column_name])

0      NaN
1      NaN
2      NaN
3      NaN
4      NaN
        ..
2842   NaN
2843   NaN
2844   NaN
2845   NaN
2846   NaN
Name: Dönen Varlıklar, Length: 2847, dtype: float64

0      NaN
1      NaN
2      NaN
3      NaN
4      NaN
        ..
2842   NaN
2843   NaN
2844   NaN
2845   NaN
2846   NaN
Name: Duran Varlıklar, Length: 2847, dtype: float64

0      NaN
1      NaN
2      NaN
3      NaN
4      NaN
        ..
2842   NaN
2843   NaN
2844   NaN
2845   NaN
2846   NaN
Name: Kısa Vadeli Yükümlülükler, Length: 2847, dtype: float64

0      NaN
1      NaN
2      NaN
3      NaN
4      NaN
        ..
2842   NaN
2843   NaN
2844   NaN
2845   NaN
2846   NaN
Name: Uzun Vadeli Yükümlülükler, Length: 2847, dtype: float64

Özkaynaklar  Özkaynaklar
0             NaN          NaN
1             NaN          NaN
2             NaN          NaN
3             NaN          NaN
4             NaN          NaN
...           ...          ...
2842          NaN          NaN
2843          NaN          NaN
2844          NaN          NaN
2845          NaN          NaN
2846          NaN          NaN

[2847 rows x 2 columns]

0      NaN
1      NaN
2      NaN
3      NaN
4      NaN
        ..
2842   NaN
2843   NaN
2844   NaN
2845   NaN
2846   NaN
Name: Ortak Kontrole Tabi Teşebbüs veya İşletmeleri İçeren Birleşmelerin Etkisi, Length: 2847, dtype: float64

0      NaN
1      NaN
2      NaN
3      NaN
4      NaN
        ..
2842   NaN
2843   NaN
2844   NaN
2845   NaN
2846   NaN
Name: Finansal Varlıklar (Net), Length: 2847, dtype: float64

0      NaN
1      NaN
2      NaN
3      NaN
4      NaN
        ..
2842   NaN
2843   NaN
2844   NaN
2845   NaN
2846   NaN
Name: Unnamed: 1, Length: 2847, dtype: float64

In [222]:
new_wide_df = wide.dropna(thresh = 1, axis  = 1)

In [224]:
new_wide_df

Company    index  Nakit ve Nakit Benzerleri  \
0      ENERY   2024/9               1.053510e+09   
1      ENERY   2024/6               6.314942e+08   
2      ENERY   2024/3               5.336559e+09   
3      ENERY  2023/12               5.935501e+09   
4      ENERY   2023/9               3.798841e+09   
...      ...      ...                        ...   
2842   AKBNK   2017/6               3.982363e+10   
2843   AKBNK   2017/3               3.326419e+10   
2844   AKBNK  2016/12               3.501228e+10   
2845   AKBNK   2016/9               3.466362e+10   
2846   AKBNK   2016/6               3.508406e+10   

      Gayrimenkul Projeleri Kapsamında Açılan Nakit Hesapları  \
0                                                   NaN         
1                                                   NaN         
2                                                   NaN         
3                                                   NaN         
4                                                   NaN         
...                                                 ...         
2842                                                NaN         
2843                                                NaN         
2844                                                NaN         
2845                                                NaN         
2846                                                NaN         

      Finansal Yatırımlar  Teminata Verilen Finansal Varlıklar  \
0                     NaN                                  NaN   
1                     NaN                                  NaN   
2                     NaN                                  NaN   
3                     NaN                                  NaN   
4                     NaN                                  NaN   
...                   ...                                  ...   
2842                  NaN                                  NaN   
2843                  NaN                                  NaN   
2844                  NaN                                  NaN   
2845                  NaN                                  NaN   
2846                  NaN                                  NaN   

      Ticari Alacaklar  Finans Sektörü Faaliyetlerinden Alacaklar  \
0         1.067704e+09                                        NaN   
1         9.063402e+08                                        NaN   
2         2.382966e+09                                        NaN   
3         3.202226e+09                                        NaN   
4         7.723428e+08                                        NaN   
...                ...                                        ...   
2842               NaN                                        NaN   
2843               NaN                                        NaN   
2844               NaN                                        NaN   
2845               NaN                                        NaN   
2846               NaN                                        NaN   

      Türkiye Cumhuriyet Merkez Bankası Hesabı  Diğer Alacaklar  \
0                                          NaN     4.542064e+06   
1                                          NaN     4.656865e+06   
2                                          NaN     1.915186e+07   
3                                          NaN     2.074794e+07   
4                                          NaN     6.369779e+07   
...                                        ...              ...   
2842                                       NaN              NaN   
2843                                       NaN              NaN   
2844                                       NaN              NaN   
2845                                       NaN              NaN   
2846                                       NaN              NaN   

      Müşteri Sözleşmelerinden Doğan Varlıklar  \
0                                          NaN   
1                                          NaN   
2                     

In [230]:
new_wide_df.columns[new_wide_df.isnull().all()]


Index([], dtype='object')

# Boş olan sütunlar temizlendi 
Sonuç olarak boş sütun kalmamış  fakat çoğunluğu boş olan satırlar ve kopyalanmış satırlar var. 
## ToDO 
*Satırların anlamlı hale gelmesini sağla 
*bir yüzde belirle bu yüzdede boş olan satırları bul 

In [257]:
new_columns = new_wide_df.columns
new_columns = [cols.split(".")[0] for cols in new_columns]

#yeni sütun adlarını df atama 
# Yeni sütun adlarını DataFrame'e ata
new_wide_df.columns = new_columns

# Sonuç kontrolü
print("Güncellenmiş Sütun Adları:")
display(new_wide_df.head(5))


Güncellenmiş Sütun Adları:


Company    index  Nakit ve Nakit Benzerleri  \
0   ENERY   2024/9               1.053510e+09   
1   ENERY   2024/6               6.314942e+08   
2   ENERY   2024/3               5.336559e+09   
3   ENERY  2023/12               5.935501e+09   
4   ENERY   2023/9               3.798841e+09   

   Gayrimenkul Projeleri Kapsamında Açılan Nakit Hesapları  \
0                                                NaN         
1                                                NaN         
2                                                NaN         
3                                                NaN         
4                                                NaN         

   Finansal Yatırımlar  Teminata Verilen Finansal Varlıklar  Ticari Alacaklar  \
0                  NaN                                  NaN      1.067704e+09   
1                  NaN                                  NaN      9.063402e+08   
2                  NaN                                  NaN      2.382966e+09   
3                  NaN                                  NaN      3.202226e+09   
4                  NaN                                  NaN      7.723428e+08   

   Finans Sektörü Faaliyetlerinden Alacaklar  \
0                                        NaN   
1                                        NaN   
2                                        NaN   
3                                        NaN   
4                                        NaN   

   Türkiye Cumhuriyet Merkez Bankası Hesabı  Diğer Alacaklar  \
0                                       NaN     4.542064e+06   
1                                       NaN     4.656865e+06   
2                                       NaN     1.915186e+07   
3                                       NaN     2.074794e+07   
4                                       NaN     6.369779e+07   

   Müşteri Sözleşmelerinden Doğan Varlıklar  \
0                                       NaN   
1                                       NaN   
2                                       NaN   
3                                       NaN   
4                                       NaN   

   İmtiyaz Sözleşmelerine İlişkin Finansal Varlıklar  Türev Araçlar  \
0                                                NaN            NaN   
1                                                NaN            NaN   
2                                                NaN            NaN   
3                                                NaN            NaN   
4                                                NaN            NaN   

        Stoklar  Proje Halindeki Stoklar  Canlı Varlıklar  \
0  1.637036e+08                      NaN              NaN   
1  1.453414e+08                      NaN              NaN   
2  1.520314e+08                      NaN              NaN   
3  1.114765e+08                      NaN              NaN   
4  8.713464e+07                      NaN              NaN   

   Peşin Ödenmiş Giderler  Ertelenmiş Sigortacılık Üretim Giderleri  \
0            5.374246e+07                                       NaN   
1            8.901621e+07                                       NaN   
2            9.346963e+07                                       NaN   
3            8.787659e+07                                       NaN   
4            2.632007e+07                                       NaN   

   Cari Dönem Vergisiyle İlgili Varlıklar  \
0                            2.852359e+08   
1                            1.615829e+07   
2                            2.714606e+05   
3                            3.639490e+07   
4                            6.790750e+06   

   Nakit Dışı Serbest Kullanılabilir Teminatlar  Diğer Dönen Varlıklar  \
0                                           NaN           1.603489e+08   
1                                           NaN           6.418253e+07   
2                                           NaN           3.669375e+08   
3                                           NaN           4.456101e+08   
4                           

In [271]:
# şimdi satır adlarındaki düzenlemeleri yaptık aynı isimde satır adları var mı bunu inceleyelim 
duplicated_columns = new_wide_df.columns.duplicated()
duplicate_names = new_wide_df.columns[duplicated_columns]
# Tekrar eden sütun isimlerini yazdır
print("Tekrar Eden Sütun İsimleri:")
print(duplicate_names)
#Tekrar eden sütun sayısı yazdırılır 
print("Tekrar eden sütun sayısı : ")
len(duplicate_names)

Tekrar Eden Sütun İsimleri:
Index(['Finansal Yatırımlar', 'Ticari Alacaklar',
       'Finans Sektörü Faaliyetlerinden Alacaklar', 'Diğer Alacaklar',
       'Müşteri Sözleşmelerinden Doğan Varlıklar',
       'İmtiyaz Sözleşmelerine İlişkin Finansal Varlıklar', 'Türev Araçlar',
       'Stoklar', 'Canlı Varlıklar', 'Peşin Ödenmiş Giderler',
       'Nakit Dışı Serbest Kullanılabilir Teminatlar', 'Türev Araçlar',
       'Finansal Borçlar', 'Diğer Finansal Yükümlülükler', 'Ticari Borçlar',
       'Finans Sektörü Faaliyetlerinden Borçlar',
       'Çalışanlara Sağlanan Faydalar Kapsamında Borçlar', 'Diğer Borçlar',
       'Müşteri Sözleşmelerinden Doğan Yükümlülükler',
       'Devlet Teşvik ve Yardımları',
       'Özkaynak Yöntemiyle Değerlenen Yatırımlardan Yükümlülükler',
       'Türev Araçlar', 'Devlet Teşvik ve Yardımları', 'Ertelenmiş Gelirler',
       'Beklenen Zarar Karşılıkları (-)', 'İlişkili Taraflardan Alacaklar',
       'Esas Faaliyetlerden Borçlar', 'İlişkili Taraflara Borçlar',
 

32

## Tekrar eden sütünlar aynı mı 
Tekrar eden sütun isimlerini bulduk şimdi bu sütünların içeriklerinin aynı olup olmadığını kıyaslamamız gerekiyor
Bunu yapmak için yöntemler araştırmak lazım 

In [281]:
#Sadece tekrar eden sütunların olduğu bir df oluşturalım 
df_deneme = new_wide_df[duplicate_names]
df_deneme.head(10)
    

Finansal Yatırımlar  Finansal Yatırımlar  Ticari Alacaklar  \
0                  NaN                  NaN      1.067704e+09   
1                  NaN                  NaN      9.063402e+08   
2                  NaN                  NaN      2.382966e+09   
3                  NaN                  NaN      3.202226e+09   
4                  NaN                  NaN      7.723428e+08   
5                  NaN                  NaN      1.200150e+09   
6                  NaN                  NaN               NaN   
7                  NaN                  NaN      5.646616e+09   
8                  NaN                  NaN      8.359033e+08   
9                  NaN                  NaN      4.417910e+08   

   Ticari Alacaklar  Finans Sektörü Faaliyetlerinden Alacaklar  \
0               NaN                                        NaN   
1               NaN                                        NaN   
2               NaN                                        NaN   
3               NaN                                        NaN   
4               NaN                                        NaN   
5               NaN                                        NaN   
6               NaN                                        NaN   
7               NaN                                        NaN   
8               NaN                                        NaN   
9               NaN                                        NaN   

   Finans Sektörü Faaliyetlerinden Alacaklar  Diğer Alacaklar  \
0                                        NaN     4.542064e+06   
1                                        NaN     4.656865e+06   
2                                        NaN     1.915186e+07   
3                                        NaN     2.074794e+07   
4                                        NaN     6.369779e+07   
5                                        NaN     2.756520e+07   
6                                        NaN              NaN   
7                                        NaN     5.593593e+07   
8                                        NaN     4.446199e+08   
9                                        NaN     5.759283e+06   

   Diğer Alacaklar  Müşteri Sözleşmelerinden Doğan Varlıklar  \
0     2.119318e+09                                       NaN   
1     2.114511e+09                                       NaN   
2     6.934067e+05                                       NaN   
3     7.031480e+05                                       NaN   
4     5.200840e+05                                       NaN   
5     5.157130e+05                                       NaN   
6              NaN                                       NaN   
7     1.008324e+06                                       NaN   
8     1.393874e+06                                       NaN   
9     1.368490e+06                                       NaN   

   Müşteri Sözleşmelerinden Doğan Varlıklar  \
0                                       NaN   
1                                       NaN   
2                                       NaN   
3                                       NaN   
4                                       NaN   
5                                       NaN   
6                                       NaN   
7                                       NaN   
8                                       NaN   
9                                       NaN   

   İmtiyaz Sözleşmelerine İlişkin Finansal Varlıklar  \
0                                                NaN   
1                                                NaN   
2                                                NaN   
3                                                NaN   
4                                                NaN   
5                                                NaN   
6                                                NaN   
7                                                NaN   
8                                                NaN   
9                                                NaN   

   İ

In [285]:
df_deneme.describe()

Finansal Yatırımlar  Finansal Yatırımlar  Ticari Alacaklar  \
count         1.480000e+03         1.621000e+03      2.346000e+03   
mean          3.542957e+09         7.530879e+09      4.621289e+09   
std           1.462263e+10         4.624775e+10      1.271097e+10   
min           0.000000e+00         0.000000e+00      0.000000e+00   
25%           4.674750e+06         7.920000e+05      3.886040e+08   
50%           1.172773e+08         6.902538e+07      1.311010e+09   
75%           1.046149e+09         8.694560e+08      3.801673e+09   
max           1.573110e+11         5.451912e+11      2.088391e+11   

       Ticari Alacaklar  Finans Sektörü Faaliyetlerinden Alacaklar  \
count      9.890000e+02                               2.790000e+02   
mean       6.188933e+08                               6.839585e+10   
std        3.381386e+09                               1.963226e+11   
min        0.000000e+00                               0.000000e+00   
25%        1.721778e+05                               0.000000e+00   
50%        6.569570e+06                               1.257035e+09   
75%        1.266690e+08                               4.785724e+09   
max        4.294456e+10                               1.020229e+12   

       Finans Sektörü Faaliyetlerinden Alacaklar  Diğer Alacaklar  \
count                               2.610000e+02     2.424000e+03   
mean                                3.820103e+10     5.981251e+08   
std                                 8.698683e+10     2.625794e+09   
min                                 0.000000e+00    -2.660036e+07   
25%                                 0.000000e+00     6.871956e+06   
50%                                 1.000970e+08     4.254615e+07   
75%                                 1.805588e+09     1.998905e+08   
max                                 4.767679e+11     4.600600e+10   

       Diğer Alacaklar  Müşteri Sözleşmelerinden Doğan Varlıklar  \
count     2.112000e+03                              2.300000e+02   
mean      3.723477e+08                              6.828373e+08   
std       2.387075e+09                              1.233305e+09   
min       0.000000e+00                              0.000000e+00   
25%       4.123900e+05                              0.000000e+00   
50%       3.640590e+06                              1.041080e+08   
75%       3.310575e+07                              8.641472e+08   
max       4.333400e+10                              7.021629e+09   

       Müşteri Sözleşmelerinden Doğan Varlıklar  \
count                              1.000000e+02   
mean                               2.444328e+07   
std                                4.528320e+07   
min                                0.000000e+00   
25%                                0.000000e+00   
50%                                0.000000e+00   
75%                                2.626020e+07   
max                                2.133265e+08   

       İmtiyaz Sözleşmelerine İlişkin Finansal Varlıklar  \
count                                       7.500000e+01   
mean                                        1.293378e+09   
std                                         2.413106e+09   
min                                         0.000000e+00   
25%                                         0.000000e+00   
50%                                         0.000000e+00   
75%                                         1.220274e+09   
max                                         9.581519e+09   

       İmtiyaz Sözleşmelerine İlişkin Finansal Varlıklar  Türev Araçlar  \
count                                       7.800000e+01   1.266000e+03   
mean                                        4.703463e+09   6.862447e+08   
std                                         7.847172e+09   2.762439e+09   
min                                         0.000000e+00   0.000000e+00   
25%                                         0.000000e+00   5.485422e+05   
50%                                         0.000000e+00   1.643821e